scripts/signup.js

```diff
@@ -0,0 +1,114 @@
+document.addEventListener("DOMContentLoaded", function () {
+  const emailInput = document.getElementById("email");
+  const nicknameInput = document.getElementById("nickname");
+  const passwordInput = document.getElementById("password");
+  const passwordConfirmInput = document.getElementById("passwordConfirmation");
+  const signupForm = document.querySelector(".auth-form");
+  const signupButton = signupForm.querySelector('button[type="submit"]');
+  const togglePasswordButtons = document.querySelectorAll(".toggle-password");
+
+  emailInput.addEventListener("blur", function () {
+    if (!isNotEmpty(this.value)) {
+      showError(this, "이메일을 입력해주세요.");
+    } else if (!isValidEmail(this.value)) {
+      showError(this, "잘못된 이메일 형식입니다.");
+    } else {
+      removeError(this);
+    }
+    validateForm();
+  });
+
+  nicknameInput.addEventListener("blur", function () {
+    if (!isNotEmpty(this.value)) {
+      showError(this, "닉네임을 입력해주세요.");
+    } else {
+      removeError(this);
+    }
+    validateForm();
+  });
+
+  passwordInput.addEventListener("blur", function () {
+    if (!isNotEmpty(this.value)) {
+      showError(this, "비밀번호를 입력해주세요.");
+    } else if (!isValidPasswordLength(this.value)) {
+      showError(this, "비밀번호를 8자 이상 입력해주세요.");
+    } else {
+      removeError(this);
+    }
+    validatePasswordMatch();
+    validateForm();
+  });
+
+  passwordConfirmInput.addEventListener("blur", function () {
+    validatePasswordMatch();
+    validateForm();
+  });
+
+  function validatePasswordMatch() {
+    if (isNotEmpty(passwordConfirmInput.value)) {
+      if (!isMatching(passwordInput.value, passwordConfirmInput.value)) {
+        showError(passwordConfirmInput, "비밀번호가 일치하지 않습니다.");
+      } else {
+        removeError(passwordConfirmInput);
+      }
+    }
+  }
+
+  // 입력 변화 감지 -> 폼 유효성 검사
+  emailInput.addEventListener("input", validateForm);
+  nicknameInput.addEventListener("input", validateForm);
+  passwordInput.addEventListener("input", validateForm);
+  passwordConfirmInput.addEventListener("input", validateForm);
+
+  function validateForm() {
+    const isEmailValid =
+      isNotEmpty(emailInput.value) && isValidEmail(emailInput.value);
+    const isNicknameValid = isNotEmpty(nicknameInput.value);
+    const isPasswordValid =
+      isNotEmpty(passwordInput.value) &&
+      isValidPasswordLength(passwordInput.value);
+    const isPasswordConfirmValid =
+      isNotEmpty(passwordConfirmInput.value) &&
+      isMatching(passwordInput.value, passwordConfirmInput.value);
+
+    const hasErrors =
+      document.querySelectorAll('[class*="text-error"]').length > 0;
+
+    setButtonState(
+      signupButton,
+      isEmailValid &&
+        isNicknameValid &&
+        isPasswordValid &&
+        isPasswordConfirmValid &&
+        !hasErrors
+    );
+  }
+
+  signupForm.addEventListener("submit", function (e) {
+    e.preventDefault();
+
+    if (!signupButton.disabled) {
+      window.location.href = "login.html";
+    }
```

Comment:
💊 제안  
form이 유효하지 않으면 submit button이 disabled 상태라 제출이 되지 않으므로 관련 코드는 없어도 될 것 같아요~

```suggestion
      window.location.href = "login.html";
```